lib/kdl/include/kdl/vector_utils.h

```diff
@@ -65,6 +65,23 @@ namespace kdl {
         return v[index_s];
     }
 
+    /**
+     * Removes the last element of the given vector and returns it.
+     *
+     * If the given vector is empty, the behavior is undefined.
+     *
+     * @tparam T the type of the vector elements
+     * @param v the vector
+     * @return the last element of the given vector
+     */
+    template <typename T>
+    T vec_pop_back(std::vector<T>& v) {
+        assert(!v.empty());
+        T result = std::move(v.back());
+        v.pop_back();
+        return result;
+    }
```

Comment:
I know this was my suggestion but UB as the default if the vec is empty feels wrong. How about this being called   `vec_pop_back_unchecked` and adding a `vec_pop_back` (throws if vector is empty).

Reply:
In this type of library code I generally do not do a lot of these checks aside from asserts. The reason is that I want it to be somewhat in line with the principles of the stl, which also avoids these kinds of checks if possible. If I change this approach here, or introduce checked vs. unchecked versions of this function, I'd have to do this for a lot of other functions in the library, too.

I will be more careful to state the preconditions of the function more clearly in the future, and have added them to the documentation where they were missing now.

Reply:
Makes sense!